formats/log/note.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"errors"
+	"fmt"
+
+	"golang.org/x/mod/sumdb/note"
+)
+
+// ParseCheckpoint takes a raw checkpoint as bytes and returns a parsed checkpoint
+// providing that a valid log signature is found, the checkpoint unmarshals
+// correctly, and the log origin is that expected. In all other cases, an empty
+// checkpoint is returned. The underlying note is always returned where possible.
```

Comment:
Any reason not to return `*Checkpoint` (nil on error)?

Reply:
Because early code reviews on my code written in Go led me to believe that pointers shouldn't be returned unless there's a good reason for it. I don't know who to believe any more. I suspect this is one of those things where everyone else has an opinion, and as someone who just wants an easy life here I'm the one that really loses out :-)

nil would be more natural to me, so I'll change it. But you better have my back when the avoid-pointers brigade turn up.

Reply:
No worries, I'll just leave you with this bit of paper telling you where I'll be if you to find need me later on... :dash: